selfservice/strategy/oidc/strategy_settings.go

```diff
@@ -550,3 +551,47 @@ func (s *Strategy) Link(ctx context.Context, i *identity.Identity, credentialsCo
 
 	return nil
 }
+
+func (s *Strategy) CompletedLogin(sess *session.Session, data *flow.DuplicateCredentialsData) error {
+	var credentialsOIDCConfig identity.CredentialsOIDC
+	if err := json.Unmarshal(data.CredentialsConfig, &credentialsOIDCConfig); err != nil {
+		return err
+	}
+	if len(credentialsOIDCConfig.Providers) != 1 {
+		return errors.New("no oidc provider was set")
```

Comment:
What if this is 2?

Reply:
it is an invalid state then, we ensure it is always 1.

Reply:
here: https://github.com/ory/kratos/blob/be5277680908f33f5a6db09e6e88e4dbbc7aeab9/selfservice/strategy/oidc/strategy_settings.go#L584